force-app/main/default/classes/SOQL.cls

```diff
@@ -16,343 +16,463 @@
-	 * Example: if there are two Contacts on the Account, "Billy" and "Abby," the 'Primary_Contact__c' will be set to "Abby."
-	 */
-	public static void setPrimaryContact() {
-		//Use the getOpportunityDetailsForClosedWon() method to get the Opportunities
-
-		// Create a set of Account Ids from the Opportunities
-
-		// Query the Contacts
-
-		// Loop through each Contact and add the Account Id and Contact to the map
-
-		// Loop through each Opportunity
-
-		// Update the Opportunity primary contact from accountToContactMap
-
-		// Update the Opportunities
-	}
+ public with sharing class SOQL {
+
+    /**
+     * Question 1
+     * Returns all leads from Salesforce Org
+     * @return List<Lead> List of all leads with Id, LastName, Company, Phone, Email
+     */
+    public static List<Lead> getAllLeads() {
+        try {
+            return [SELECT Id, LastName, Company, Phone, Email
+                    FROM Lead
+                    WHERE LastName != null
+                    LIMIT 50000];
+        } catch(Exception e) {
+            return new List<Lead>();
+        }
+    }
+
+    /**
+     * Question 2
+     * Returns all contacts with first name of John
+     * @return List<Contact> A list of Contact records
+     */
+    public static List<Contact> getContactsByFirstName() {
+        try {
+            return [SELECT Id, FirstName, LastName, Name
+                    FROM Contact
+                    WHERE FirstName = 'John'
+                    AND LastName != null
+                    LIMIT 50000];
+        } catch(Exception e) {
+            return new List<Contact>();
+        }
+    }
+
+    /**
+     * Question 3
+     * Returns Contact Ids for contacts with specific last names
+     * @return List<Id> A list of Contact Ids
+     */
+    public static List<Id> getContactIdsByLastName() {
+        try {
+            List<Id> contactIds = new List<Id>();
+            for(Contact c : [SELECT Id
+                           FROM Contact
+                           WHERE LastName IN ('Smith', 'Jones')
+                           AND LastName != null
+                           LIMIT 50000]) {
+                contactIds.add(c.Id);
+            }
+            return contactIds;
+        } catch(Exception e) {
+            return new List<Id>();
+        }
+    }
+
+    /**
+     * Question 4
+     * Returns List of closed won opportunities for a specific Account
+     * @param accountId The ID of the Account
+     * @return List<Opportunity> List of filtered opportunities
+     */
+    public static List<Opportunity> getClosedWonOpportunityCountByAccount(Id accountId) {
+        try {
+            if(accountId == null) {
+                return new List<Opportunity>();
+            }
+            return [SELECT Id, Name, Amount, CloseDate, AccountId, stageName
+                    FROM Opportunity
+                    WHERE AccountId = :accountId
+                    AND StageName = 'Closed Won'
+                    AND Amount != null
```

Comment:
Also very hard to create an opp without an amount, so this null check can be removed.